api/v1/cluster_webhook.go

```diff
@@ -1539,6 +1550,21 @@ func (r *Cluster) validateTablespacesNames() field.ErrorList {
 	}
 
 	for name := range r.Spec.Tablespaces {
+		if slices.Contains(reservedTablespaceName, name) {
+			result = append(result, field.Invalid(
+				field.NewPath("spec", "tablespaces"),
+				name,
+				"tablespace name must not be a reserved name"))
+		}
+
+		if !tablespaceNameRegex.MatchString(name) {
+			result = append(result, field.Invalid(
+				field.NewPath("spec", "tablespaces"),
+				name,
+				"tablespace name must consist of lower case alphanumeric characters,'-' or '.', "+
+					"and must start and end with an alphanumeric character"))
+		}
```

Comment:
I don't think we should do this. In Postgres, tablespace names don't have to obey this. We need transform functions to convert back and forth for K8s labels and such.

Reply:
I agree we should remove this restriction. This is the pvc name restriction, definitely will remove it. as we put the tablespace name into the label of the pvc, we can use the regular expression to valid the converted pvc name.